components/hab/src/cli.rs

```diff
@@ -803,3 +807,10 @@ fn valid_url(val: String) -> result::Result<(), String> {
         Err(_) => Err(format!("URL: '{}' is not valid", &val)),
     }
 }
+
+fn valid_numeric<T: FromStr>(val: String) -> result::Result<(), String> {
```

Comment:
I think you might be able to omit this function and use the [value_t](https://docs.rs/clap/2.31.2/clap/macro.value_t.html) macro when parsing.

Reply:
Yeah, It's also nice to have the validation up front, similar to the url validation, so it seemed better to follow that pattern